src/NHibernate.Test/LazyOneToOne/LazyOneToOneTest.cs

```diff
@@ -67,8 +67,8 @@ public void Lazy()
 
 			s = OpenSession();
 			t = s.BeginTransaction();
-			p = s.Get<Person>("Gavin");
-			Assert.That(!NHibernateUtil.IsPropertyInitialized(p, "Employee"));
+			p = s.Get<Person>("Gavin"); // The default loader will fetch the employee
+			Assert.That(NHibernateUtil.IsPropertyInitialized(p, "Employee"));
```

Comment:
Isn't that a regression? `Employee` property was supposed to be lazy, but it looks to be no more.

Reply:
The problem here is that the current `IsPropertyInitialized` does not work correctly for this case as the `Employeee` is eager loaded when the `Person` is retrieved, but as the interceptor is currently injected after the entity initialization it doesn't know that the `Employee` was already initialized in the 2-phase load. If the `Employee` wasn't meant to be eager loaded (by using a join) when retrieving the `Persion`, then it should be treated as a different issue.

Reply:
Do you mean that for some reason, `no-proxy` on non-inverse `one-to-one` is broken since long and always eager load the employee anyway? (But an additional bug was causing this eager load to be "unseen" by the property.)